### ISSUE TITLE
[Bug] Error in rule patch_segmentation_baysor

### ISSUE THREAD

**Issue by AnaisJulien** [gustaveroussy/sopa#118]
Hey Quentin, 

Thanks for te very nice tolls you developed. 

I'm trying t run it on Merscope data using the Snakemake pipeline, using a custom .yaml file based on your baysor_vizgen.yaml one. 

After sometimes, I get the following error: 

" Error in rule patch_segmentation_baysor:
    jobid: 36
    input: /Volumes/Merscope_1/Slide_2/merfish_output/Sample1/region_3.zarr/.sopa_cache/patches_file_baysor, /Volumes/Merscope_1/Slide_2/merfish_output/Sample1/region_3.zarr/.sopa_cache/baysor_boundaries/53
    output: /Volumes/Merscope_1/Slide_2/merfish_output/Sample1/region_3.zarr/.sopa_cache/baysor_boundaries/53/segmentation_polygons.json, /Volumes/Merscope_1/Slide_2/merfish_output/Sample1/region_3.zarr/.sopa_cache/baysor_boundaries/53/segmentation_counts.loom
    shell:
        
        if command -v module &> /dev/null; then
            module purge
        fi

        cd /Volumes/Merscope_1/Slide_2/merfish_output/Sample1/region_3.zarr/.sopa_cache/baysor_boundaries/53
        ~/.julia/bin/baysor run --save-polygons GeoJSON -c config.toml transcripts.csv :cell_id
        
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!) "

It looks like for other jobid it's working well. 

Do you have an idea on what is the problem ? 

Thank you for your help, 

Anais

**Comment by quentinblampey** [comment]
Hello @AnaisJulien, do you have the error log for this specific job id? I see `jobid: 36` in your logs, and this job should have its specific log file

For instance, if you use the SLURM profile, then the log is inside `sopa/workflow/logs`!

**Comment by AnaisJulien** [comment]
Hi, 

It seems like the logs folder is empty.

**Comment by quentinblampey** [comment]
But did you use the SLURM profile? Can you show me the snakemake command you used to run the pipeline?

**Comment by AnaisJulien** [comment]
I did not use the SLURM profile. 
I used the following command:  anajul@KI-C02D223BMD6T workflow % snakemake --config data_path=/Volumes/Merscope_1/Slide_2/merfish_output/Slide1/region_3 --configfile=/Volumes/Merscope_1/Slide_2/baysor_vizgen.yaml --cores 8 --use-conda

**Comment by came203** [comment]
Dear all,

I had a similar problem a couple of days ago. For me, our HPC admin crew does not allow us to run the module purge command, so that one fails. I commented it out, ran fine afterwards

**Comment by AnaisJulien** [comment]
Thank you fro your answer. 
Could you provide the code you used please ? 

Thanks

**Comment by came203** [comment]
In your sopa_directory/workflow/Snakefile:
```

rule patch_segmentation_baysor:
    input:
        patches_file = paths.smk_patches_file_baysor,
        baysor_patch = paths.smk_baysor_temp_dir / "{index}",
    output:
        paths.smk_baysor_temp_dir / "{index}" / "segmentation_polygons.json",
        paths.smk_baysor_temp_dir / "{index}" / "segmentation_counts.loom",
    params:
        args_baysor_prior_seg = args.baysor_prior_seg,
    resources:
        mem_mb=128_000,
    shell:
        """
        #if command -v module &> /dev/null; then
        #    module purge
        #fi

        cd {input.baysor_patch}
        {config[executables][baysor]} run --save-polygons GeoJSON -c config.toml transcripts.csv {params.args_baysor_prior_seg}
```

So just commenting out the if clause did it for me. Hope that helps

**Comment by AnaisJulien** [comment]
Hey, 

Thanks for your help but it did not work. I still get the same error...

**Comment by quentinblampey** [comment]
I'm closing the issue as it was resolved outside of this feed.

The issue was related to the baysor installation. The Baysor command itself (e.g., `~/.julia/bin/baysor --help`) was returning the error below:
```
ERROR: System image file failed consistency check: maybe opened the wrong version?
```
This error thus comes from the Baysor installation. Some potential fixes: installing Baysor with Julia 1.8 or opening an issue on the Baysor repository.
